discovery/gossiper.go

```diff
@@ -343,22 +344,27 @@ func New(cfg Config, selfKey *btcec.PublicKey) *AuthenticatedGossiper {
 // specified, then the update will be applied to all outgoing channels from the
 // source node. Policy updates are done in two stages: first, the
 // AuthenticatedGossiper ensures the update has been committed by dependent
-// sub-systems, then it signs and broadcasts new updates to the network.
+// sub-systems, then it signs and broadcasts new updates to the network. A
+// mapping between outpoints and updated channel policies is returned, which is
+// used to update the forwarding policies of the underlying links.
 func (d *AuthenticatedGossiper) PropagateChanPolicyUpdate(
-	newSchema routing.ChannelPolicy, chanPoints ...wire.OutPoint) error {
+	newSchema routing.ChannelPolicy, chanPoints ...wire.OutPoint) (
+	map[wire.OutPoint]*channeldb.ChannelEdgePolicy, error) {
 
 	errChan := make(chan error, 1)
+	chanPolicyChan := make(chan map[wire.OutPoint]*channeldb.ChannelEdgePolicy, 1)
 	policyUpdate := &chanPolicyUpdateRequest{
-		targetChans: chanPoints,
-		newSchema:   newSchema,
-		errResp:     errChan,
+		targetChans:  chanPoints,
+		newSchema:    newSchema,
+		chanPolicies: chanPolicyChan,
+		errResp:      errChan,
 	}
 
 	select {
 	case d.chanPolicyUpdates <- policyUpdate:
-		return <-errChan
+		return <-chanPolicyChan, <-errChan
```

Comment:
Can these chan reads block forever because there is no select on `d.quit`? I think in the error case, nothing is sent on `chanPolicyChan`.